### PR TITLE
V1 ansible scripts

### PR DESCRIPTION
**Pull request by RangerMauve**
These scripts will make it easy for others to run their own version of DP on any systemd-based Linux server over SSH.

TODO:

- [x] Inventory file to deploy to staging environment
- [x] Docs on how to run this
- [x] Testing on staging environment
- [ ] Code review by Ansible wizards

### CODE REVIEW

File: ansible/README.md
Comment (jackyzha0) [inline ansible/README.md:11]:
```suggestion
Dependencies:
```

File: ansible/README.md
Comment (jackyzha0) [inline ansible/README.md:18]:
```suggestion
# execute the Ansible playbook with user defined variables
ansible-playbook distributed_press.yml -i inventory.yml
```

File: ansible/roles/distributed_press/templates/distributed.press.service.j2
Comment (jackyzha0) [inline ansible/roles/distributed_press/templates/distributed.press.service.j2:10]: Do we need to bump this? (see: https://www.freedesktop.org/software/systemd/man/systemd.exec.html#Process%20Properties) Seems like potentially unsafe to bump beyond 1024

File: ansible/roles/distributed_press/tasks/main.yml
Comment (jackyzha0) [inline ansible/roles/distributed_press/tasks/main.yml:40]: We should add a few TODO comments that point out what needs to change when v1 is eventually merged (e.g. changing the branch)

As a separate question: do we eventually want to get rid of the v0 folder as well?

File: ansible/roles/distributed_press/tasks/main.yml
Comment (jackyzha0) [inline ansible/roles/distributed_press/tasks/main.yml:42]:
```suggestion
    npm ci
```

**Review by jackyzha0 — CHANGES_REQUESTED** [review]
Left a few comments! Rest looks good :)

Ansible cool

File: ansible/roles/distributed_press/templates/distributed.press.service.j2
Comment (RangerMauve) [reply ansible/roles/distributed_press/templates/distributed.press.service.j2:10]: Not sure actually. @hyphacoop/infrastructure any thoughts?

File: ansible/roles/distributed_press/tasks/main.yml
Comment (RangerMauve) [reply ansible/roles/distributed_press/tasks/main.yml:40]: Yeah, I feel like it'd be safe to get rid of v0 as part of merging v1 in.

Maybe in the README we could point to the last stable v0 release as a tag for those that need it (our legacy deploys I think).

Maybe we can get to that once we merge v1 in to the main branch?

**Review by jackyzha0 — APPROVED** [review]
Tentative approve! Probably wait for infra to review but looks good to me :)

File: ansible/roles/distributed_press/templates/distributed.press.service.j2
Comment (darkdrgn2k) [reply ansible/roles/distributed_press/templates/distributed.press.service.j2:10]: I think the key in the comment I think your referring to is 
> Typically applications should increase their soft limit to the hard limit on their own
Which I would assume the app does not do.

 I don't see the danger other then not using `select(2)` and possible resource utilization (DDOS). In un-managed code I can see leaks of not closing descriptors when your done with them, but with a GC I don't think its much of a danger if leakage.

I have seen this value set much higher in other apps esp distributed systems I have seen this value set much higher then 4096. For example in Saturn the docker container is set to 10,000 (but i guess that is system wide not for a single process)
https://github.com/filecoin-saturn/L1-node/blob/main/docker-compose.yml#L12-L15

Keep in mind this is the `Distributed Press API Server` its going to be touching lots of file descriptors, form incoming sockets, to outgoing connections to reading files and metadata.

Just my 2 cents.

File: ansible/roles/distributed_press/tasks/main.yml
Comment (YurkoWasHere) [reply ansible/roles/distributed_press/tasks/main.yml:40]: I think the key in the comment I think your referring to is 
> Typically applications should increase their soft limit to the hard limit on their own
Which I would assume the app does not do.

 I don't see the danger other then not using `select(2)` and possible resource utilization (DDOS). In un-managed code I can see leaks of not closing descriptors when your done with them, but with a GC I don't think its much of a danger if leakage.

I have seen this value set much higher in other apps esp distributed systems I have seen this value set much higher then 4096. For example in Saturn the docker container is set to 10,000 (but i guess that is system wide not for a single process)
https://github.com/filecoin-saturn/L1-node/blob/main/docker-compose.yml#L12-L15

Keep in mind this is the `Distributed Press API Server` its going to be touching lots of file descriptors, form incoming sockets, to outgoing connections to reading files and metadata.

Just my 2 cents.

File: ansible/roles/distributed_press/templates/distributed.press.service.j2
Comment (RangerMauve) [reply ansible/roles/distributed_press/templates/distributed.press.service.j2:10]: K, let's keep it as is then and revisit it if it's a problem. :+1:

File: ansible/roles/distributed_press/tasks/main.yml
Comment (RangerMauve) [reply ansible/roles/distributed_press/tasks/main.yml:40]: Gonna leave this a s a TODO for when we merge v1-staging into the main branch.

**Pull request merged**
